apps/internal/oauth/ops/authority/authority.go

```diff
@@ -166,48 +166,7 @@ func firstPathSegment(u *url.URL) (string, error) {
 // NewInfoFromAuthorityURI creates an AuthorityInfo instance from the authority URL provided.
 func NewInfoFromAuthorityURI(authorityURI string, validateAuthority bool) (Info, error) {
 	authorityURI = strings.ToLower(authorityURI)
-
-	// TODO(msal): The original code I found had a validateAuthority() function.
-	// But the function just returned error == nil. I took this from the python and
-	// it looks like this needs a lot more. Right now we just pretend its validated.
-	/*
-			if (tenant != "adfs" and (not is_b2c) and validate_authority
-		                and self.instance not in WELL_KNOWN_AUTHORITY_HOSTS):
-		            payload = instance_discovery(
-		                "https://{}{}/oauth2/v2.0/authorize".format(
-		                    self.instance, authority.path),
-		                self.http_client)
-		            if payload.get("error") == "invalid_instance":
-		                raise ValueError(
-		                    "invalid_instance: "
-		                    "The authority you provided, %s, is not whitelisted. "
-		                    "If it is indeed your legit customized domain name, "
-		                    "you can turn off this check by passing in "
-		                    "validate_authority=False"
-		                    % authority_url)
-		            tenant_discovery_endpoint = payload['tenant_discovery_endpoint']
-		        else:
-		            tenant_discovery_endpoint = (
-		                'https://{}{}{}/.well-known/openid-configuration'.format(
-		                    self.instance,
-		                    authority.path,  # In B2C scenario, it is "/tenant/policy"
-		                    "" if tenant == "adfs" else "/v2.0" # the AAD v2 endpoint
-		                    ))
-		        try:
-		            openid_config = tenant_discovery(
-		                tenant_discovery_endpoint,
-		                self.http_client)
-		        except ValueError:
-		            raise ValueError(
-		                "Unable to get authority configuration for {}. "
-		                "Authority would typically be in a format of "
-		                "https://login.microsoftonline.com/your_tenant_name".format(
-						authority_url))
-	*/
-
-	// todo: check for other authority types...
-	authorityType := "MSSTS"
-
+	var authorityType string
```

Comment:
We do this in `resolvers.go`  [here](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/dev/apps/internal/oauth/resolvers.go#L48)
We delay doing this here because we want to avoid making a network call unless absolutely needed. So this code wont run while initializing app object here.